centreon/src/Core/HostCategory/Infrastructure/Repository/DbReadHostCategoryRepository.php

```diff
@@ -583,4 +619,41 @@ private function hasRestrictedAccessToHostCategories(array $accessGroupIds): boo
 
         return (bool) $statement->fetchColumn();
     }
+
+    /**
+     * Determine if host groups are filtered for given access group ids
+     * true: accessible host groups are filtered (only specified are accessible)
+     * false: accessible host groups are not filtered (all are accessible).
+     *
+     * @param int[] $accessGroupIds
+     *
+     * @phpstan-param non-empty-array<int> $accessGroupIds
+     *
+     * @return bool
+     */
+    private function hasRestrictedAccessToHostGroups(array $accessGroupIds): bool
+    {
+        $concatenator = new SqlConcatenator();
+
+        $concatenator->defineSelect(
+            'SELECT 1
+            FROM `:db`.acl_resources_hg_relations arhgr
+            INNER JOIN `:db`.acl_resources res
+                ON arhgr.acl_res_id = res.acl_res_id
+            INNER JOIN `:db`.acl_res_group_relations argr
+                ON res.acl_res_id = argr.acl_res_id
+            INNER JOIN `:db`.acl_groups ag
+                ON argr.acl_group_id = ag.acl_group_id'
+        );
+
+        $concatenator->storeBindValueMultiple(':access_group_ids', $accessGroupIds, \PDO::PARAM_INT)
+            ->appendWhere('ag.acl_group_id IN (:access_group_ids)');
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->__toString()));
+
+        $concatenator->bindValuesToStatement($statement);
+        $statement->execute();
+
+        return (bool) $statement->fetchColumn();
+    }
```

Comment:
Host group ACLs doesn't work on a "filter" system like the categories, but on the UI you can check a checkbox that gives you access to all host group. You need to check if this option with the hasAccessToAllHostGroups method

```suggestion
```